src/Config/ObjectModel/RuntimeConfig.cs

```diff
@@ -44,28 +47,90 @@ public IEnumerable<DataSource> ListAllDataSources()
     /// </summary>
     /// <param name="Schema">schema.</param>
     /// <param name="DataSource">Default datasource.</param>
     /// <param name="Runtime">Runtime settings.</param>
     /// <param name="Entities">Entities</param>
+    /// <param name="DataSourceFiles">List of datasource files for multiple db scenario.</param></param>
     [JsonConstructor]
-    public RuntimeConfig(string Schema, DataSource DataSource, RuntimeOptions Runtime, RuntimeEntities Entities)
+    public RuntimeConfig(string Schema, DataSource DataSource, RuntimeOptions Runtime, RuntimeEntities Entities, DataSourceFiles? DataSourceFiles = null)
     {
         this.Schema = Schema;
         this.DataSource = DataSource;
         this.Runtime = Runtime;
         this.Entities = Entities;
-        this._dataSourceNameToDataSource = new Dictionary<string, DataSource>();
         this._defaultDataSourceName = Guid.NewGuid().ToString();
-        this._dataSourceNameToDataSource.Add(this._defaultDataSourceName, this.DataSource);
+
+        // we will set them up with default values
+        this._dataSourceNameToDataSource = new Dictionary<string, DataSource>
+        {
+            { this._defaultDataSourceName, this.DataSource }
+        };
 
         this._entityNameToDataSourceName = new Dictionary<string, string>();
+
         foreach (KeyValuePair<string, Entity> entity in Entities)
         {
             _entityNameToDataSourceName.TryAdd(entity.Key, this._defaultDataSourceName);
         }
 
+        // Multiple database scenario.
+        this.DataSourceFiles = DataSourceFiles;
+        IEnumerable<KeyValuePair<string, Entity>> allEntities = Entities.AsEnumerable();
+
+        if (DataSourceFiles is not null)
+        {
+            try
+            {
+                // Iterate through all the datasource files and load the config.
+                IFileSystem fileSystem = new FileSystem();
+                FileSystemRuntimeConfigLoader loader = new(fileSystem);
+
+                foreach (string dataSourceFile in DataSourceFiles.SourceFiles ?? Enumerable.Empty<string>())
+                {
+                    if (loader.TryLoadConfig(dataSourceFile, out RuntimeConfig? config))
+                    {
+                        this._dataSourceNameToDataSource = this._dataSourceNameToDataSource.Concat(config._dataSourceNameToDataSource).ToDictionary(kvp => kvp.Key, kvp => kvp.Value);
+                        this._entityNameToDataSourceName = this._entityNameToDataSourceName.Concat(config._entityNameToDataSourceName).ToDictionary(kvp => kvp.Key, kvp => kvp.Value);
+                        allEntities = allEntities.Concat(config.Entities.ToList());
+                    }
+                }
+
+                this.Entities = new RuntimeEntities(allEntities.ToDictionary(x => x.Key, x => x.Value));
+            }
+            catch (Exception e)
+            {
+                // Errors could include invalid sub file paths, duplicated entity names, etc.
```

Comment:
Is there a challenge right now in identifying the file which probably contained a duplicate entity name? I see that you are concatenating entity details to the dictionary and if there is a duplicate entity name, we will get a general c# error. Having the file name which contains the duplicate entityname would be a real help to the user.

Reply:
I see that we need to store the source file names for both entities which have duplicate names, so this might be more involved.

Reply:
So the actual validation for this will come from the CLI and will be part of when the runtimeConfig's are generated. you will have a validate that can validate all your runtimeConfigs for the multi-db scenario there. However, i moved the try catch logic to the part where we actually try to merge the datasource files so you know the datasource file that has the issue. If file path is not present, we are expected to move ahead and continue rather than error out - this accounts for the static web app scenario.